### ISSUE TITLE
Kobuki Gazebo Plugin

### ISSUE THREAD

**Issue by stonier** [yujinrobot/kobuki#156]
Wait until groovy before we do this.

**Comment by bit-pirate** [comment]
Tried to get the Kobuki simulation starting in Groovy, but gave up after a while. Looking through the available documentation out there made me confused. E.g. documentation on the ROS wiki looks quite outdated and documentation on the Gazebosim wiki does not clearly point out, which the correct way of using Gazebo in Groovy is.
There is also the issue of the model description file format. Some people are starting to create extra SDF description (the default Gazebo format). To avoid wasting time, I asked a [question on answers.gazebosim.org](http://answers.gazebosim.org/question/782/recommended-way-of-using-gazebo-with-ros-groovy/). I'll continue the struggle after I get some answer there.

**Comment by stonier** [comment]
I'll ping Tully and Brian on the recommended way to do gazebo on groovy.

**Comment by bit-pirate** [comment]
OK, I got a bit further:
- We have a scaling issue with our collada meshes. RViz is not unit-aware, but Gazebo now is. I a hack to fix it, but the cool looking surfaces are still messed up. DK is now checking, how to export correct collada files.
- Strange position&orientation bug I had with Gazebo some monts ago is gone. When I start up Turtlebot, it looks well aligned, only the Kinect seems a bit off. Have to investigate what's the cause for this.
- Gazebo ROS plugins: I had to disable them to get things starting. So, updating them is still TODO.
- URDF vs. SDF: This still seems to be an open issue and I don't understand, why this hasn't been solved already or at least discussed more. Gazebo is widely used among ROS users and this is a big issue. Gazebo is going it's own way with the more versatile SDF file format for describing everything in the simulation world. They already have a online library from which people can download models and just drag&drop them into the simulation world. However, it is not supported in the ROS world (RViz, robot_state_publisher and tf (the latter two I'm assuming). There is a URDF to SDF parser, which I don't think is well maintained. So, some people are starting to create extra SDF models of their robots, what I really would like to avoid.

I'll try to find a way to get a plugin-less simulation of Kobuki and Turtlebot working and will then have a look at the plugins. The ugly meshes (misaligned surface images) will be replaced as soon as DK is able to fix them. And maybe I'll try to raise some attention for the SDF issue.
